src/sql-parser/src/ast/defs/statement.rs

```diff
@@ -569,6 +569,7 @@ pub struct CreateSourceStatement<T: AstInfo> {
     pub with_options: Vec<CreateSourceOption<T>>,
     pub referenced_subsources: Option<ReferencedSubsources<T>>,
     pub progress_subsource: Option<DeferredItemName<T>>,
+    pub upsert_order_by: Option<Ident>,
```

Comment:
Makes sense, let me open a design doc PR (and it's going be pretty brief 😅 ).

OOC when we say it's an option of upsert, because otherwise it's more like feature/option on the source itself and we want to make it explicit?

Reply:
(Design doc here https://github.com/MaterializeInc/materialize/pull/18722)